partitioncraft/application.py

```diff
@@ -0,0 +1,40 @@
+# This file is part of craft_application.
+#
+# Copyright 2025 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License version 3, as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+# SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License along
+# with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Application implementation for partitioncraft."""
+
+import craft_cli
+import craft_parts
+from typing_extensions import override
+
+import craft_application
+
+PARTITIONCRAFT = craft_application.AppMetadata(
```

Comment:
nitpick: AFAICS in every other *craft applications it is named `APP_METADATA`. In testcraft it is named `TESTCRAFT`. I have no strong opinion on either solution (generic or per-application name) but I think we should keep it consistent, so either rename `TESTCRAFT` and `PARTITIONCRAFT` to `APP_METADATA` or change it in our *craft applications.